src/librustc_mir_build/build/matches/mod.rs

```diff
@@ -1039,6 +1047,53 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         reachable_candidates.last_mut().unwrap().otherwise_block
     }
 
+    /// Tests a candidate where there are only or-patterns left to test, or
+    /// forwards to [Builder::test_candidates].
+    ///
+    /// Given a pattern `(P | Q, R | S)` we (in principle) generate a CFG like
```

Comment:
the edges in the below control-flow graph are all going down, right?

For some reason, the bullet that reads "If there's a guard, then the otherwise branch of the first match on  `R | S` goes to a test for whether `Q` matches." made me start wondering if I was misunderstanding the graph, and that led me to realize that this was a source of slight ambiguity.